From 7491ee6a5a5e9c72c5154022a7151a18a86599cb (util-linux)
From: Petr Uzel <petr.uzel@suse.cz>
Date: Mon, 14 Feb 2011 14:43:23 +0100
Subject: [PATCH] libuuid: fix test in get_clock()

Signed-off-by: Petr Uzel <petr.uzel@suse.cz>
---

--- a/lib/uuid/gen_uuid.c
+++ b/lib/uuid/gen_uuid.c
@@ -392,7 +392,7 @@ try_again:
 		last.tv_usec = last.tv_usec % 1000000;
 	}
 
-	if (state_fd > 0) {
+	if (state_fd >= 0) {
 		rewind(state_f);
 		len = fprintf(state_f,
 			      "clock: %04x tv: %016lu %08lu adj: %08d\n",
